tests/mir-opt/issue_99325.main.built.after.32bit.mir

```diff
@@ -2,7 +2,7 @@
 
 | User Type Annotations
 | 0: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[d56d]::function_with_bytes), UserArgs { args: [&*b"AAAA"], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:13:16: 13:46, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
-| 1: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[d56d]::function_with_bytes), UserArgs { args: [UnevaluatedConst { def: DefId(0:8 ~ issue_99325[d56d]::main::{constant#1}), args: [] }], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:14:16: 14:68, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
+| 1: user_ty: Canonical { value: TypeOf(DefId(0:3 ~ issue_99325[d56d]::function_with_bytes), UserArgs { args: [&*b"AAAA"], user_self_ty: None }), max_universe: U0, variables: [] }, span: $DIR/issue_99325.rs:14:16: 14:68, inferred_ty: fn() -> &'static [u8] {function_with_bytes::<&*b"AAAA">}
```

Comment:
This change would be why we *dont* want to normalize at the end of `lower_const_arg`. User type annotations exist to remember the "unnormalized as written by the user" types/generic arguments in MIR so that any lifetime constraints introduced by them can be known to the borrow checker instead of being ignored :3

This *specific* example isn't wrong per se but in the general case it would be with const arguments that use lifetime parameters, e.g.:
```rust
#![feature(min_generic_const_args)]

trait Trait<'a> {
  const ASSOC: usize;
}

impl Trait<'static> for u8 {
  const ASSOC: usize = 10;
}

struct Bar<const N: usize>;

fn foo<'a>() {
  let _: Bar<<u8 as Trait<'a>>::ASSOC>;
}
```

`<u8 as Trait<'a>>::ASSOC` being well formed requires `'a: 'static` which doesn't hold so we *should* error in borrow checking. This is accomplished by having a user type annotation of `<u8 as Trait<'a>>::ASSOC` so that borrow checking knows `'a: 'static` must hold. With this change we'd have a user type annotation of `10` which doesn't tell borrow checking anything about `'a: 'static` having to hold :3

Though this example won't actually do anything right now since `min_generic_const_args` hasn't been implemented :3